core/system_impl.cpp

```diff
@@ -187,11 +187,13 @@ void SystemImpl::process_heartbeat(const mavlink_message_t &message)
     if (is_autopilot(message.compid) && !have_uuid()) {
         request_autopilot_version();
 
+#if defined(ENABLE_FALLBACK_TO_SYSTEM_ID)
```

Comment:
Not in the scope of this PR (because it affects other parts), but as a general comment: I'm always nervous with `#ifdef`s. We could design this to use a boolean instead, couldn't we?

Reply:
You're right.